sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -123,15 +123,16 @@ public static Write write() {
   public abstract static class Write extends PTransform<PCollection<Mutation>, PDone> {
 
     @Nullable
-    abstract String getProjectId();
+    abstract ValueProvider<String> getProjectId();
 
     @Nullable
-    abstract String getInstanceId();
+    abstract ValueProvider<String> getInstanceId();
 
     @Nullable
-    abstract String getDatabaseId();
+    abstract ValueProvider<String> getDatabaseId();
 
-    abstract long getBatchSizeBytes();
+    @Nullable
+    abstract ValueProvider<Long> getBatchSizeBytes();
```

Comment:
Okay, though consider making this a simpler-to-use builder method .withoutBatching().

Also: do you expect people to actually configure this via a ValueProvider? (I don't think we have guidelines on what builder methods should have a ValueProvider overload, but if we had such a guideline, it probably wouldn't be "all of them")